queries/luap/highlights.scm

```diff
@@ -1,9 +1,11 @@
-"." @character
-
 [
   (anchor_begin)
   (anchor_end)
-] @string.escape
+  "."
+] @variable.builtin
+
+((character "." @constant) @_
+ (#has-parent? @_ set negated_set))
```

Comment:
The comment hours prior from me was weird.

For this use case, I like two queries rather using predicates. It generates less captures than the current version

```suggestion
(set
  (character "." @constant))
(negated_set
  (character "." @constant)) 
```

Reply:
Sounds good, thanks for the input! While we're at it, I noticed this query:
```scheme
(negated_set
  "^" @operator
  (character) @constant)
```
highlights `"^"` as `@operator` for each character matched in the negated set, e.g. `[^ffffffffff]` will have the `^` highlighted as (per `:Inspect`):
```
Treesitter
  - @string.lua links to String lua
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  - @operator.luap links to Operator luap
  ```
Do you think this should also be split into two queries? Or is this not much of an issue

Reply:
Another follow up; I think I may have to add the following query if it's alright:
```scheme
(range
  (character "." @constant))
```
Right now for something like `[.-0]`, the `.` is still highlighted as `@variable.builtin` rather than `@constant`. I am remembering now that this is why I had the ancestor query originally

Reply:
> Do you think this should also be split into two queries? Or is this not much of an issue

Yes it should. This should not be happening for any language/query

Reply:
> Another follow up; I think I may have to add the following query if it's alright:
> 
> ```scheme
> 
> (range
> 
>   (character "." @constant))
> 
> ```
> 
> Right now for something like `[.-0]`, the `.` is still highlighted as `@variable.builtin` rather than `@constant`. I am remembering now that this is why I had the ancestor query originally

What about the reverse?

```scheme
(character "." @constant) 
; Top-level dot
(pattern
  (character "." @variable.builtin))
```

Reply:
Oh nice! This is great

Reply:
Let me know if I should squash all of these commits if there are now too many